Testing/integration/SNTExecTest.m

```diff
@@ -0,0 +1,66 @@
+/// Copyright 2021 Google Inc. All rights reserved.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///    http://www.apache.org/licenses/LICENSE-2.0
+///
+///    Unless required by applicable law or agreed to in writing, software
+///    distributed under the License is distributed on an "AS IS" BASIS,
+///    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+///    See the License for the specific language governing permissions and
+///    limitations under the License.
+#import <Foundation/Foundation.h>
+#import <XCTest/XCTest.h>
+
+#include <unistd.h>
+
+@interface SNTExecTest : XCTestCase
+@end
+
+@implementation SNTExecTest : XCTestCase
+- (void)setUp {
+  [super setUp];
+  fclose(stdout);
+}
+
+- (void)checkExecution:(NSString *)path shouldExec:(BOOL)shouldExec {
+  __block int status;
+
+  XCTestExpectation *expectation =
+    [self expectationWithDescription:@"Wait for test binary to execute"];
+
+  dispatch_async(dispatch_get_global_queue(QOS_CLASS_BACKGROUND, 0), ^{
+    NSTask *task = [[NSTask alloc] init];
+    task.launchPath = path;
+    [task launch];
+    while ([task isRunning])
+      usleep(100000);  // 0.1s
```

Comment:
```suggestion
    [task waitUntilExit];
```

Shouldn't it just be this? Then if your expectation isn't met within the alloted time you can just kill it with `[task terminate];`

Reply:
Done